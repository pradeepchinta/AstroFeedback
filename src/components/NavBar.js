
import React from "react";
import { Navbar, Nav, Form, FormControl, Button } from "react-bootstrap";

const NavBar = ({ handleSearch, showSearch, searchText, handleChange, searchEnabled }) => {
    return (
        <Navbar bg="light" expand="lg">
            <Navbar.Brand href="/">Astro Customer Feedbacks</Navbar.Brand>
            <Navbar.Toggle aria-controls="basic-navbar-nav" />
            <Navbar.Collapse id="basic-navbar-nav">
                <Nav className="mr-auto">
                    <Nav.Link href="/add-new">Add New Feedback</Nav.Link>
                </Nav>
                {showSearch && <Form inline>
                    <FormControl type="text" placeholder="Search Using Email" className="mr-sm-2" value={searchText} onChange={handleChange}/>
                    <Button id="SearchButton" variant="outline-success" onClick={handleSearch} disabled={searchEnabled} >Search</Button>
                </Form>}
            </Navbar.Collapse>
        </Navbar>
    )
};

export default NavBar;