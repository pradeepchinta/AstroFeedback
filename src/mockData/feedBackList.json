[
    {
      "id": "1",
      "title": "title 1",
      "type": "type 1",
      "description": "description 1",
      "email": "email 1",
      "date": 1593590610,
      "like": 19,
      "dislike": 88
    },
    {
      "id": "2",
      "title": "title 2",
      "type": "type 2",
      "description": "description 2",
      "email": "email 2",
      "date": 1593590550,
      "like": 80,
      "dislike": 9
    },
    {
      "id": "3",
      "title": "title 3",
      "type": "type 3",
      "description": "description 3",
      "email": "email 3",
      "date": 1593590490,
      "like": 10,
      "dislike": 27
    },
    {
      "id": "4",
      "title": "title 4",
      "type": "type 4",
      "description": "description 4",
      "email": "email 4",
      "date": 1593590430,
      "like": 39,
      "dislike": 41
    }
]