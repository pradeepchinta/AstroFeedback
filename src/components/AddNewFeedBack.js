import React, { useState, useRef, useEffect } from "react";
import { Container, Alert, Form, Button } from "react-bootstrap";
import moment from 'moment';
import { connect } from 'react-redux';

import NavBar from "./NavBar";
import { saveFeedback, hideMessage, showMessage } from '../actions';

export const NewFeedBack = ({ saveFeedback, responseMessage, responseVariant, hideMessage, showMessage }) => {
    const [validated, setValidated] = useState(false);
    const formRef = useRef(null);

    useEffect(() => {
        setTimeout(() => {
            hideMessage();
            if (responseMessage === "FeedBack Created Sucessfully") {
                formRef.current.reset();
                setValidated(false);
            }
        }, 2000)
    }, [responseMessage])

    const renderAlert = () => {
        return (
            <Alert variant={responseVariant}>
                {responseMessage}
            </Alert>
        )
    };

    const handleSubmit = (event) => {
        event.preventDefault();
        const form = event.currentTarget;
        const reg = /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/;
        if (form.checkValidity() === true) {
            const data = {
                title: form.formBasicTitle.value,
                type: form.formBasicType.value,
                description: form.formBasicDescription.value,
                email: form.formBasicEmail.value,
                date: moment(),
                like: 0,
                disLike: 0
            };
            if (reg.test(form.formBasicEmail.value)) {
                saveFeedback(data);
            } else {
                showMessage({
                    message: "Please enter valid email ...!",
                    variant: "danger"
                })
            }
        }
        setValidated(true);
    };



    return (
        <Container fluid>
            <NavBar showSearch={false} />
            <Form ref={formRef} noValidate validated={validated} onSubmit={handleSubmit}>
                <Form.Group controlId="formBasicTitle">
                    <Form.Label>Title</Form.Label>
                    <Form.Control required type="title" placeholder="Title" />
                    <Form.Control.Feedback type="invalid">Please Provide Feedback title</Form.Control.Feedback>
                </Form.Group>
                <Form.Group controlId="formBasicType">
                    <Form.Label>Type</Form.Label>
                    <Form.Control required as="select">
                        <option value="">-- Select --</option>
                        <option value="Technical">Technical</option>
                        <option value="User">User</option>
                        <option value="Management Related">Management Related</option>
                        <option value="competitor Related">competitor Related</option>
                        <option value="Pricing issue">Pricing issue</option>
                        <option value="Customer Care issue">Customer Care issue</option>
                    </Form.Control>
                    <Form.Control.Feedback type="invalid">Please Select Feedback type</Form.Control.Feedback>
                </Form.Group>
                <Form.Group controlId="formBasicEmail">
                    <Form.Label>Email address</Form.Label>
                    <Form.Control required type="email" placeholder="Enter email" onChange={hideMessage} />
                    <Form.Control.Feedback type="invalid">Please Enter valid Email</Form.Control.Feedback>
                </Form.Group>
                <Form.Group controlId="formBasicDescription">
                    <Form.Label>Description</Form.Label>
                    <Form.Control required as="textarea" rows="3" />
                    <Form.Control.Feedback type="invalid">Please Enter Feedback Description</Form.Control.Feedback>
                </Form.Group>
                {(responseMessage) && renderAlert()}
                <Button variant="primary" type="submit">
                    Submit
                </Button>
            </Form>
        </Container>
    )
};

const mapStateToProps = (state) => {
    return {
        feedBackList: state.feedBackReducer.feedBackList,
        responseMessage: state.feedBackReducer.responseMessage,
        responseVariant: state.feedBackReducer.responseVariant
    };
};
const mapDispatchToProps = (dispatch) => {
    return {
        saveFeedback: (data) => dispatch(saveFeedback(data)),
        hideMessage: () => dispatch(hideMessage()),
        showMessage: (data) => dispatch(showMessage(data))
    };
};
export default connect(mapStateToProps, mapDispatchToProps)(NewFeedBack);