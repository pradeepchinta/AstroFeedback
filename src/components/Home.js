import React, { useEffect, useState } from "react";
import { connect } from 'react-redux';
import moment from 'moment';
import { Container, Media, Card, Spinner, Alert, Button } from "react-bootstrap";

import NavBar from "./NavBar";
import { fetchFeedBacks, searchFeedbacks, hideMessage, showMessage, updateFeedback } from '../actions';
import personImage from '../assets/person.png';

export const Home = ({ feedBackList, fetchFeedBacks, searchFeedbacks, responseMessage, responseVariant, showMessage, hideMessage, updateFeedback }) => {

    const [searchText, setSearchText] = useState("");
    const [searchEnabled, setSearchEnabled] = useState(false);
    const [liked, setLiked] = useState(false);
    const [disliked, setDisliked] = useState(false);
    //const [like, setLike] = useState(0);
    //const [disLike, setDisLike] = useState(0);
    let like = 0;
    let disLike = 0;

    useEffect(() => {
        fetchFeedBacks();
    }, []);

    useEffect(() => {
        setTimeout(() => {
            hideMessage();
        }, 2000)
    }, [responseMessage])

    const handleSearch = () => {
        const reg = /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/;
        if (!searchText) {
            fetchFeedBacks();
            setSearchEnabled(false);
            return
        }
        if (reg.test(searchText)) {
            searchText && searchFeedbacks(searchText);
            setSearchEnabled(true);
        } else {
            showMessage({
                message: "Please enter valid email to search ...!",
                variant: "danger"
            })
            setSearchEnabled(false);
        }
    };

    const handleChange = (e) => {
        setSearchText(e.target.value);
    }

    const handleClick = (e, item) => {
        switch (e.target.name) {
            case "like":
                if (liked) {
                    like = item.like -1;
                    disLike = item.disLike;
                    setLiked(false);
                } else {
                    like = ++item.like;
                    disLike = item.disLike;
                    setLiked(true);
                }
                break;
            case "dislike":
                if (disliked) {
                    disLike = item.disLike -1;
                    like = item.like;
                    setDisliked(false);
                } else {
                    disLike = ++item.disLike;
                    like = item.like;
                    setDisliked(true);
                }
                break;
            default:
                return null;
        };
        const data = {
            id: item.id,
            title: item.title,
            type: item.type,
            description: item.description,
            email: item.email,
            date: item.date,
            like: like,
            disLike: disLike
        };
        updateFeedback(data);
    }

    const compare = (item, item1) => {
        if (item1.like > item.like) return 1;
        if (item.like > item1.like) return -1;
        return 0;
    }

    const renderList = () => {
        return feedBackList && feedBackList.sort(compare).map((item) => {
            return <div key={item.id}>
                <Media as="li">
                    <img
                        width={64}
                        height={64}
                        className="mr-3"
                        src={personImage}
                        alt="Generic placeholder"
                    />
                    <Media.Body>
                        <h5>{item.title}</h5>
                        <strong>{item.type}</strong>
                        <p>{item.description}</p>
                        <p>{item.email}</p>
                        <p>{moment(item.date).format("DD-MM-YYYY")}</p>
                        <Button id="like" onClick={(e) => handleClick(e, item)} variant="outline-primary" type="button" name="like">{`Like ${item.like}`}</Button>{' '}
                        <Button id="dislike" onClick={(e) => handleClick(e, item)} variant="outline-primary" type="button" name="dislike">{`DisLike ${item.disLike}`}</Button>
                    </Media.Body>
                </Media>
                <hr />
            </div >
        })
    };

    const renderSpinner = () => {
        return (
            <React.Fragment>
                <Spinner animation="border" role="status">
                    <span className="sr-only">Loading...</span>
                </Spinner>
                {feedBackList && !!feedBackList.length && <h2>We have Empty FeedBack List ...</h2>}
            </React.Fragment>
        )
    };

    const renderAlert = () => {
        return (
            <Alert variant={responseVariant}>
                {responseMessage}
            </Alert>
        )
    };

    return (
        <Container fluid>
            <NavBar handleSearch={handleSearch} showSearch={true} handleChange={handleChange} searchText={searchText} searchEnabled={searchEnabled}/>
            {(responseMessage && responseVariant) && renderAlert()}
            <Card>
                {feedBackList && feedBackList.length ? renderList() : (
                    <div className='spinner'>
                        {searchEnabled ? <h2>No FeedBack List Found</h2> : renderSpinner()}
                    </div>)}
            </Card>
        </Container>
    )
};

const mapStateToProps = (state) => {
    return {
        feedBackList: state.feedBackReducer.feedBackList,
        responseMessage: state.feedBackReducer.responseMessage,
        responseVariant: state.feedBackReducer.responseVariant
    };
};
const mapDispatchToProps = (dispatch) => {
    return {
        fetchFeedBacks: () => dispatch(fetchFeedBacks()),
        searchFeedbacks: (textSearch) => dispatch(searchFeedbacks(textSearch)),
        updateFeedback: (data) => dispatch(updateFeedback(data)),
        hideMessage: () => dispatch(hideMessage()),
        showMessage: (data) => dispatch(showMessage(data))
    };
};
export default connect(mapStateToProps, mapDispatchToProps)(Home);