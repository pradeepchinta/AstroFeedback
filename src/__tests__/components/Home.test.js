import React from "react";
import { mount } from "enzyme";
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import { Home } from "../../components/Home";
import feedBackList from "../../mockData/feedBackList";

const middlewares = [thunk]
const mockStore = configureStore(middlewares);
const store = mockStore({});

describe("Home Component", () => {
    let wrapper;

    const minProps = {
        feedBackList,
        responseMessage: "fetched Successfully",
        responseVariant: "success",
        fetchFeedBacks: jest.fn(),
        searchFeedbacks: jest.fn(),
        updateFeedback: jest.fn(),
        hideMessage: jest.fn(),
        showMessage: jest.fn()
    };

    it("will render with min props", () => {
        wrapper = mount(<Home {...minProps} store={store} />);
        expect(wrapper).toMatchSnapshot();
    })
    it("will render NavBar Component", () => {
        expect(wrapper.find("NavBar").length).toBe(1);
    })
    it("will render spinner when no data", () => {
        wrapper.setProps({
            feedBackList: []
        });
        expect(wrapper.find("Spinner").length).toBe(1);
    })
    it(" will render feedbackList with most liked one first", () => {
        wrapper.setProps({
            feedBackList
        });
        expect(wrapper.find("h5").first().props().children).toBe("title 2");
    })
})