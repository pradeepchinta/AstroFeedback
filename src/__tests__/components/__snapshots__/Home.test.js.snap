// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component will render with min props 1`] = `
<Home
  feedBackList={
    Array [
      Object {
        "date": 1593590550,
        "description": "description 2",
        "dislike": 9,
        "email": "email 2",
        "id": "2",
        "like": 80,
        "title": "title 2",
        "type": "type 2",
      },
      Object {
        "date": 1593590430,
        "description": "description 4",
        "dislike": 41,
        "email": "email 4",
        "id": "4",
        "like": 39,
        "title": "title 4",
        "type": "type 4",
      },
      Object {
        "date": 1593590610,
        "description": "description 1",
        "dislike": 88,
        "email": "email 1",
        "id": "1",
        "like": 19,
        "title": "title 1",
        "type": "type 1",
      },
      Object {
        "date": 1593590490,
        "description": "description 3",
        "dislike": 27,
        "email": "email 3",
        "id": "3",
        "like": 10,
        "title": "title 3",
        "type": "type 3",
      },
    ]
  }
  fetchFeedBacks={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  hideMessage={[MockFunction]}
  responseMessage="fetched Successfully"
  responseVariant="success"
  searchFeedbacks={[MockFunction]}
  showMessage={[MockFunction]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  updateFeedback={[MockFunction]}
>
  <Container
    fluid={true}
  >
    <div
      className="container-fluid"
    >
      <NavBar
        handleChange={[Function]}
        handleSearch={[Function]}
        searchEnabled={false}
        searchText=""
        showSearch={true}
      >
        <Navbar
          bg="light"
          collapseOnSelect={false}
          expand="lg"
          variant="light"
        >
          <nav
            className="navbar navbar-expand-lg navbar-light bg-light"
          >
            <NavbarBrand
              href="/"
            >
              <a
                className="navbar-brand"
                href="/"
              >
                Astro Customer Feedbacks
              </a>
            </NavbarBrand>
            <NavbarToggle
              aria-controls="basic-navbar-nav"
              label="Toggle navigation"
            >
              <button
                aria-controls="basic-navbar-nav"
                aria-label="Toggle navigation"
                className="navbar-toggler collapsed"
                onClick={[Function]}
                type="button"
              >
                <span
                  className="navbar-toggler-icon"
                />
              </button>
            </NavbarToggle>
            <NavbarCollapse
              id="basic-navbar-nav"
            >
              <Collapse
                appear={false}
                dimension="height"
                getDimensionValue={[Function]}
                id="basic-navbar-nav"
                in={false}
                mountOnEnter={false}
                timeout={300}
                unmountOnExit={false}
              >
                <Transition
                  addEndListener={[Function]}
                  appear={false}
                  aria-expanded={null}
                  enter={true}
                  exit={true}
                  id="basic-navbar-nav"
                  in={false}
                  mountOnEnter={false}
                  onEnter={[Function]}
                  onEntered={[Function]}
                  onEntering={[Function]}
                  onExit={[Function]}
                  onExited={[Function]}
                  onExiting={[Function]}
                  timeout={300}
                  unmountOnExit={false}
                >
                  <div
                    aria-expanded={null}
                    className="navbar-collapse collapse"
                    id="basic-navbar-nav"
                  >
                    <Nav
                      className="mr-auto"
                      fill={false}
                      justify={false}
                    >
                      <ForwardRef
                        as="div"
                        className="mr-auto navbar-nav"
                        onSelect={[Function]}
                      >
                        <div
                          className="mr-auto navbar-nav"
                          onKeyDown={[Function]}
                        >
                          <NavLink
                            as={
                              Object {
                                "$$typeof": Symbol(react.forward_ref),
                                "displayName": "SafeAnchor",
                                "render": [Function],
                              }
                            }
                            disabled={false}
                            href="/add-new"
                          >
                            <ForwardRef
                              as={
                                Object {
                                  "$$typeof": Symbol(react.forward_ref),
                                  "displayName": "SafeAnchor",
                                  "render": [Function],
                                }
                              }
                              className="nav-link"
                              disabled={false}
                              href="/add-new"
                            >
                              <SafeAnchor
                                className="nav-link"
                                data-rb-event-key="/add-new"
                                disabled={false}
                                href="/add-new"
                                onClick={[Function]}
                              >
                                <a
                                  className="nav-link"
                                  data-rb-event-key="/add-new"
                                  href="/add-new"
                                  onClick={[Function]}
                                  onKeyDown={[Function]}
                                >
                                  Add New Feedback
                                </a>
                              </SafeAnchor>
                            </ForwardRef>
                          </NavLink>
                        </div>
                      </ForwardRef>
                    </Nav>
                    <Form
                      inline={true}
                    >
                      <form
                        className="form-inline"
                      >
                        <FormControl
                          className="mr-sm-2"
                          onChange={[Function]}
                          placeholder="Search Using Email"
                          type="text"
                          value=""
                        >
                          <input
                            className="mr-sm-2 form-control"
                            onChange={[Function]}
                            placeholder="Search Using Email"
                            type="text"
                            value=""
                          />
                        </FormControl>
                        <Button
                          active={false}
                          disabled={false}
                          id="SearchButton"
                          onClick={[Function]}
                          type="button"
                          variant="outline-success"
                        >
                          <button
                            className="btn btn-outline-success"
                            disabled={false}
                            id="SearchButton"
                            onClick={[Function]}
                            type="button"
                          >
                            Search
                          </button>
                        </Button>
                      </form>
                    </Form>
                  </div>
                </Transition>
              </Collapse>
            </NavbarCollapse>
          </nav>
        </Navbar>
      </NavBar>
      <Alert
        closeLabel="Close alert"
        show={true}
        transition={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": Object {
              "appear": false,
              "in": false,
              "mountOnEnter": false,
              "timeout": 300,
              "unmountOnExit": false,
            },
            "displayName": "Fade",
            "render": [Function],
          }
        }
        variant="success"
      >
        <Fade
          appear={false}
          in={true}
          mountOnEnter={false}
          timeout={300}
          unmountOnExit={true}
        >
          <Transition
            addEndListener={[Function]}
            appear={false}
            enter={true}
            exit={true}
            in={true}
            mountOnEnter={false}
            onEnter={[Function]}
            onEntered={[Function]}
            onEntering={[Function]}
            onExit={[Function]}
            onExited={[Function]}
            onExiting={[Function]}
            timeout={300}
            unmountOnExit={true}
          >
            <div
              className="fade alert alert-success show"
              role="alert"
            >
              fetched Successfully
            </div>
          </Transition>
        </Fade>
      </Alert>
      <Card
        body={false}
      >
        <div
          className="card"
        >
          <div
            key="2"
          >
            <Media
              as="li"
            >
              <li
                className="media"
              >
                <img
                  alt="Generic placeholder"
                  className="mr-3"
                  height={64}
                  src="person.png"
                  width={64}
                />
                <MediaBody>
                  <div
                    className="media-body"
                  >
                    <h5>
                      title 2
                    </h5>
                    <strong>
                      type 2
                    </strong>
                    <p>
                      description 2
                    </p>
                    <p>
                      email 2
                    </p>
                    <p>
                      19-01-1970
                    </p>
                    <Button
                      active={false}
                      disabled={false}
                      id="like"
                      name="like"
                      onClick={[Function]}
                      type="button"
                      variant="outline-primary"
                    >
                      <button
                        className="btn btn-outline-primary"
                        disabled={false}
                        id="like"
                        name="like"
                        onClick={[Function]}
                        type="button"
                      >
                        Like 80
                      </button>
                    </Button>
                     
                    <Button
                      active={false}
                      disabled={false}
                      id="dislike"
                      name="dislike"
                      onClick={[Function]}
                      type="button"
                      variant="outline-primary"
                    >
                      <button
                        className="btn btn-outline-primary"
                        disabled={false}
                        id="dislike"
                        name="dislike"
                        onClick={[Function]}
                        type="button"
                      >
                        DisLike undefined
                      </button>
                    </Button>
                  </div>
                </MediaBody>
              </li>
            </Media>
            <hr />
          </div>
          <div
            key="4"
          >
            <Media
              as="li"
            >
              <li
                className="media"
              >
                <img
                  alt="Generic placeholder"
                  className="mr-3"
                  height={64}
                  src="person.png"
                  width={64}
                />
                <MediaBody>
                  <div
                    className="media-body"
                  >
                    <h5>
                      title 4
                    </h5>
                    <strong>
                      type 4
                    </strong>
                    <p>
                      description 4
                    </p>
                    <p>
                      email 4
                    </p>
                    <p>
                      19-01-1970
                    </p>
                    <Button
                      active={false}
                      disabled={false}
                      id="like"
                      name="like"
                      onClick={[Function]}
                      type="button"
                      variant="outline-primary"
                    >
                      <button
                        className="btn btn-outline-primary"
                        disabled={false}
                        id="like"
                        name="like"
                        onClick={[Function]}
                        type="button"
                      >
                        Like 39
                      </button>
                    </Button>
                     
                    <Button
                      active={false}
                      disabled={false}
                      id="dislike"
                      name="dislike"
                      onClick={[Function]}
                      type="button"
                      variant="outline-primary"
                    >
                      <button
                        className="btn btn-outline-primary"
                        disabled={false}
                        id="dislike"
                        name="dislike"
                        onClick={[Function]}
                        type="button"
                      >
                        DisLike undefined
                      </button>
                    </Button>
                  </div>
                </MediaBody>
              </li>
            </Media>
            <hr />
          </div>
          <div
            key="1"
          >
            <Media
              as="li"
            >
              <li
                className="media"
              >
                <img
                  alt="Generic placeholder"
                  className="mr-3"
                  height={64}
                  src="person.png"
                  width={64}
                />
                <MediaBody>
                  <div
                    className="media-body"
                  >
                    <h5>
                      title 1
                    </h5>
                    <strong>
                      type 1
                    </strong>
                    <p>
                      description 1
                    </p>
                    <p>
                      email 1
                    </p>
                    <p>
                      19-01-1970
                    </p>
                    <Button
                      active={false}
                      disabled={false}
                      id="like"
                      name="like"
                      onClick={[Function]}
                      type="button"
                      variant="outline-primary"
                    >
                      <button
                        className="btn btn-outline-primary"
                        disabled={false}
                        id="like"
                        name="like"
                        onClick={[Function]}
                        type="button"
                      >
                        Like 19
                      </button>
                    </Button>
                     
                    <Button
                      active={false}
                      disabled={false}
                      id="dislike"
                      name="dislike"
                      onClick={[Function]}
                      type="button"
                      variant="outline-primary"
                    >
                      <button
                        className="btn btn-outline-primary"
                        disabled={false}
                        id="dislike"
                        name="dislike"
                        onClick={[Function]}
                        type="button"
                      >
                        DisLike undefined
                      </button>
                    </Button>
                  </div>
                </MediaBody>
              </li>
            </Media>
            <hr />
          </div>
          <div
            key="3"
          >
            <Media
              as="li"
            >
              <li
                className="media"
              >
                <img
                  alt="Generic placeholder"
                  className="mr-3"
                  height={64}
                  src="person.png"
                  width={64}
                />
                <MediaBody>
                  <div
                    className="media-body"
                  >
                    <h5>
                      title 3
                    </h5>
                    <strong>
                      type 3
                    </strong>
                    <p>
                      description 3
                    </p>
                    <p>
                      email 3
                    </p>
                    <p>
                      19-01-1970
                    </p>
                    <Button
                      active={false}
                      disabled={false}
                      id="like"
                      name="like"
                      onClick={[Function]}
                      type="button"
                      variant="outline-primary"
                    >
                      <button
                        className="btn btn-outline-primary"
                        disabled={false}
                        id="like"
                        name="like"
                        onClick={[Function]}
                        type="button"
                      >
                        Like 10
                      </button>
                    </Button>
                     
                    <Button
                      active={false}
                      disabled={false}
                      id="dislike"
                      name="dislike"
                      onClick={[Function]}
                      type="button"
                      variant="outline-primary"
                    >
                      <button
                        className="btn btn-outline-primary"
                        disabled={false}
                        id="dislike"
                        name="dislike"
                        onClick={[Function]}
                        type="button"
                      >
                        DisLike undefined
                      </button>
                    </Button>
                  </div>
                </MediaBody>
              </li>
            </Media>
            <hr />
          </div>
        </div>
      </Card>
    </div>
  </Container>
</Home>
`;
